src/Fixer/ClassNotation/ClassAttributesSeparationFixer.php

```diff
@@ -355,7 +383,39 @@ private function fixSpaceAboveClassElement(Tokens $tokens, int $classStartIndex,
             return;
         }
 
-        $this->correctLineBreaks($tokens, $nonWhiteAbove, $firstElementAttributeIndex, $nonWhiteAbove === $classStartIndex || self::SPACING_NONE === $spacing ? 1 : 2);
+        $reqLineCount = $this->determineRequiredLineCount($tokens, $nonWhiteAbove, $classStartIndex, $spacing);
+
+        $this->correctLineBreaks($tokens, $nonWhiteAbove, $firstElementAttributeIndex, $reqLineCount);
+    }
+
+    /**
+     * @param int    $notWhiteIndex
+     * @param int    $classStartOrEndIndex
+     * @param string $spacing
+     *
+     * @return int
+     */
+    private function determineRequiredLineCount(Tokens $tokens, $notWhiteIndex, $classStartOrEndIndex, $spacing)
+    {
+        // if the two indices are equal, this means the index is either at the
+        // start or end of the class and no additional line breaks are needed
+        if ($notWhiteIndex === $classStartOrEndIndex) {
+            return 1;
+        }
+
+        $lineCount = 2;
+
+        // initially, `hybrid` and `none` behaves the same
+        if (\in_array($spacing, [self::SPACING_HYBRID, self::SPACING_NONE], true)) {
+            $lineCount = 1;
```

Comment:
We can use early returns instead of a variable to improve readability:
```
if (/* ... */) {
    return 1;
}
```
This would require to change the `if`s order though.